go/oasis-node/cmd/common/metrics/metrics.go

```diff
@@ -41,28 +73,46 @@ type ServiceConfig struct {
 	Address string
 	// JobName is the name of the job for which metrics are collected.
 	JobName string
-	// InstanceLabel is the instance label of the job being collected for.
-	InstanceLabel string
+	// Labels are the key-value labels of the job being collected for.
+	Labels map[string]string
 	// Interval defined the push interval for metrics collection.
 	Interval time.Duration
 }
 
 // GetServiceConfig gets the metrics configuration parameter struct.
 func GetServiceConfig() *ServiceConfig {
 	return &ServiceConfig{
-		Mode:          viper.GetString(cfgMetricsMode),
-		Address:       viper.GetString(cfgMetricsAddr),
-		JobName:       viper.GetString(cfgMetricsPushJobName),
-		InstanceLabel: viper.GetString(cfgMetricsPushInstanceLabel),
-		Interval:      viper.GetDuration(cfgMetricsPushInterval),
+		Mode:     viper.GetString(CfgMetricsMode),
+		Address:  viper.GetString(CfgMetricsAddr),
+		JobName:  viper.GetString(CfgMetricsPushJobName),
+		Labels:   ParseMetricPushLabels(viper.GetString(CfgMetricsPushLabels)),
+		Interval: viper.GetDuration(CfgMetricsPushInterval),
 	}
 }
 
 type stubService struct {
 	service.BaseBackgroundService
+
+	dService *diskService
```

Comment:
It looks like these services are all the same, so it would be nicer if there was just an array of interfaces. This way you could consolidate duplicate code (e.g., invoking start on all the services etc.).